scripts/exhafs_forecast.sh

```diff
@@ -713,6 +714,12 @@ if [ ${warmstart_from_restart} = yes ]; then
   # currently only update the variable itself but not its checksum attribute.
   ncatted -a checksum,,d,, ${RESTARTinp}/${PDY}.${cyc}0000.fv_core.res.tile1.nc ./fv_core.res.tile1.nc
   ncatted -a checksum,,d,, ${RESTARTinp}/${PDY}.${cyc}0000.fv_tracer.res.tile1.nc ./fv_tracer.res.tile1.nc
+#
+  line2=`sed -n '2p' coupler.res`
+  line3=`sed -n '3p' coupler.res`
+  hh2=$(echo $line2 | awk '{print $4}')
+  hh3=$(echo $line3 | awk '{print $4}')
+  ${NSED} -i 's/\b'$hh2'\b/'$hh3'/' coupler.res
```

Comment:
Upon further testing, this sed command will not work if the first cycle is 0 UTC, because sed will be looking to replace all 0 (with word breaks) with 6.

If the initial coupler.res looks like this:

     2        (Calendar: no_calendar=0, thirty_day_months=1, julian=2, gregorian=3, noleap=4)
     2020     8    22     0     0     0        Model start time:   year, month, day, hour, minute, second
     2020     8    22     6     0     0        Current model time: year, month, day, hour, minute, second

The edited coupler.res will be:
```
 2        (Calendar: no_calendar=6, thirty_day_months=1, julian=2, gregorian=3, noleap=4)
 2020     8    22     6     0     0        Model start time:   year, month, day, hour, minute, second
 2020     8    22     6     6     0        Current model time: year, month, day, hour, minute, second
```
The forecast job will then fail because the minute column differs. The no_calendar variable also changes (not sure what the effect would be). Also need to be aware that for a 6 or 12 UTC cycle, the months column could get overwritten. And the days column could be changed if hh=dd.

A more minor comment is that this line uses ${NSED}, but there are many lines in the script that just use sed outright, so maybe it is not necessary to define NSED for consistency's sake.

Reply:
@evankalina and @panll, You may see how I fixed this in the feature/hafsv0.3_baseline branch as a reference:
https://github.com/hafs-community/HAFS/blob/feature/hafsv0.3_baseline/scripts/exhafs_forecast.sh#:~:text=%24%7BNLN%7D%20%24%7BRESTARTinp%7D/%24%7BYMD%7D.%24%7Bhh%7D0000.coupler.res%20./coupler.res,time%3A%20year%2C%20month%2C%20day%2C%20hour%2C%20minute%2C%20second/%22%20./coupler.res

Thanks!

Reply:
@BinLiu-NOAA thanks! @linlin-pan I just tested Bin's fix on the cycle in question and it works. If you want to see how I implemented it, you can look at this file on Jet: /mnt/lfs4/HFIP/dtc-hurr/Evan.Kalina/HAFS-Agnes/scripts/exhafs_forecast.sh. It is very straightforward. I would commit it but I don't have access to your fork.

Reply:
Thanks for the comments and suggestions! @BinLiu-NOAA @evankalina I'll modify that accordingly.